src/module/actor/creature/sheet.ts

```diff
@@ -38,7 +38,7 @@ abstract class CreatureSheetPF2e<TActor extends CreaturePF2e> extends ActorSheet
 
         sheetData.data.perception.senses = R.sortBy(sheetData.data.perception.senses, (a) => a.label ?? "");
         const initiativeOptions = [{ value: "perception", label: "PF2E.PerceptionLabel" }];
-        initiativeOptions.push(...Object.values(sheetData.data.skills).map((s) => ({ value: s.slug, label: s.label })));
+        initiativeOptions.push(...Object.values(this.actor.skills).map((s) => ({ value: s.slug, label: s.label })));
```

Comment:
Why this?

Reply:
These two things should probably be separate PRs, but both together allowed the sheet to open despite bad data prep. If skills fail data prep for any reason, there are two points of failure: numberFormat on the skills tab, and the initiative selection.

The initiative selection is a new point of failure in V12, because of the deprecation of `#select` moving us towards `#selectOptions`. This is a point though where I saw we could reasonably guarantee that the data coming in was correct.

With the true bug fixed (updated the migration) both of these can wait till next release, and we can probably fix corrupted data period as well while we're at it (only for skills).